### ISSUE TITLE
Multi platform support 

### ISSUE THREAD

**Issue by KMilhan** [deepskydetail/AstroSharp#1]
Thanks for the awesome project!

Do you have any plan to support another platform such as Linux?

**Comment by deepskydetail** [comment]
Yes, at some point! I'm currently trying to get it working for mac (not much success), also looking at Linux.

**Comment by SimonTelescopium** [comment]
Thanks for doing this. I would love to see a LINUX version also :-)

**Comment by lcoronelp** [comment]
Hello,

I have created a shell script file to eliminate the necessity of using RStudio.

This script was tested on Xubuntu 22.04 with a fresh R installation. Although the process of installing dependencies took a considerable amount of time, it successfully completed in the end.

The script is named "run.sh" and is located in the same "shiny" folder.

When executed, it opens a new tab in the default browser, displaying the running app.

Here is the script:
```bash
#!/bin/bash

# INSTALL DEPENDENCIES #

# List of packages
packages=("shiny" "shinycssloaders" "neuralnet" "dplyr" "ggthemes" "tiff" "htmltools" "reshape2" "waveslim" "imager" "bslib" "parallel" "foreach" "doParallel" "doSNOW" "remotes")

# Loop over the packages and install if not already installed
for package in "${packages[@]}"
do
    R --no-echo -e "if (!require('${package}', character.only = TRUE)) install.packages('${package}', repos='http://cran.rstudio.com/')"
done

# Install 'neuralnet' package from GitHub
R --no-echo -e "remotes::install_github('bips-hb/neuralnet')"


# RUN #

# Get the directory of the current script
dir="${0%/*}"

# Run the R script
R -e "shiny::runApp('$dir/app.R',launch.browser=TRUE)" --no-save
```

I hope this works on other operating systems like Mac or other Unix-based systems.

Greetings and many thanks again for such a good app.

PS: I may consider proposing a visual and UX rework soon, depending on the time my job allows me.

**Comment by deepskydetail** [comment]
Awesome! Thank you so much for this :)

**Comment by SimonTelescopium** [comment]
Thanks for doing this, I'm a bit of a noob, my main interest is astronomy. My system, Tuxedo OS2, doesn't have R, RStudio or shiny - I'm not even completely sure what they are but I think they are a bit like apt? Any suggestions?

**Comment by lcoronelp** [comment]
Hi, the modified script to work with the last version of the code `5.0.7` in Linux (add the `shinyWidgets` package)

```
#!/bin/bash

# INSTALL DEPENDENCIES #

# List of packages
packages=("shiny" "shinycssloaders" "neuralnet" "dplyr" "ggthemes" "tiff" "htmltools" "reshape2" "waveslim" "imager" "bslib" "parallel" "foreach" "doParallel" "doSNOW" "remotes" "shinyWidgets")

# Loop over the packages and install if not already installed
for package in "${packages[@]}"
do
    R --no-echo -e "if (!require('${package}', character.only = TRUE)) install.packages('${package}', repos='http://cran.rstudio.com/')"
done

# Install 'neuralnet' package from GitHub
R --no-echo -e "remotes::install_github('bips-hb/neuralnet')"


# RUN #

# Get the directory of the current script
dir="${0%/*}"

# Run the R script
R -e "shiny::runApp('$dir/app.R',launch.browser=TRUE)" --no-save
```

**Comment by ajeb78** [comment]
I've tested the script on Ubuntu 23.10 and it works fine, except for the last line which was trying to run /bin/app.R, however once the installation was done it was simple enough to figure out how to modify the last line to run the app.

**Comment by stevelcb** [comment]
Hi 
Great project. Thanks.

Problems getting it to run on Ubuntu 22.04.3. Running the script went OK except it was missing imager. I installed imager. At the R prompt:

install.packages("imager")

Then:

R -e "shiny::runApp('/home/steve/AstroSharp/resources/app/app.R',launch.browser=TRUE)" --no-save

No idea if this is the intended method though.

Thanks again,
Steve




```

**Comment by stevelcb** [comment]
Hi
Problem using Hybrid model
Ubuntu 22.04
Any ideas on this?
TIA,
Steve

```

Warning: Error in if: argument is of length zero
  168: renderPlot [/home/steve/AstroSharp/resources/app/app.R#552]
  166: func
  126: drawPlot
  112: <reactive:plotObj>
   96: drawReactive
   83: renderFunc
   82: output$p2
    1: shiny::runApp
Warning: Error in hist.default: 'x' must be numeric
  171: stop
  170: hist.default
  168: renderPlot [/home/steve/AstroSharp/resources/app/app.R#572]
  166: func
  126: drawPlot
  112: <reactive:plotObj>
   96: drawReactive
   83: renderFunc
   82: output$p3
    1: shiny::runApp
```

Screen:
![p2](https://github.com/deepskydetail/AstroSharp/assets/49716273/f9d8fe90-eee7-484e-be16-f52e9c6f7abc)

**Comment by stevelcb** [comment]
I've put a how-to-build-on-Ubuntu here:
https://linuxcb.blogspot.com/2024/03/deep-sky-detail-on-linux.html
But please bear in mind, I'm not an expert in anything tech. So _at your own risk_ and stuff...
Cheers

**Comment by deepskydetail** [comment]
Thanks for doing the blog post! And sorry for not getting back sooner!

With the first error, you could open up R and run the following code
"install.packages("imager"). But it looks like you figured that out already
;)

The hybrid model has been broken for a while now. I need to put out a fix.
I just haven't had much time lately. Let me know if I missed anything.

Thanks!
Mark
